libqtile/command/base.py

```diff
@@ -55,11 +57,118 @@ class CommandException(Exception):
 class CommandObject(metaclass=abc.ABCMeta):
     """Base class for objects that expose commands
 
-    Each command should be a method named `cmd_X`, where X is the command name.
+    Any command to be exposed should be decorated with
+    `@CommandObject.expose()` (classes that are not explicitly
+    inheriting from CommandObject will need to import the module)
     A CommandObject should also implement `._items()` and `._select()` methods
     (c.f. docstring for `.items()` and `.select()`).
     """
 
+    def __new__(cls, *args, **kwargs):
+
+        # Check which level of command object has been parsed
+        # This test ensures inherited classes don't stop additional
+        # methods from being exposed.
+        # For example, if widget.TextBox has already been parsed, a subsequent
+        # call to initialise a new TextBox will return here. However, if a user
+        # subclasses TextBox for a new widget then that new widget will still
+        # be parsed here to check for new commands.
+        if getattr(cls, "_command_object", "") == cls.__name__:
+            return super(CommandObject, cls).__new__(cls)
+
+        commands = {}
+        cmd_s = set()
+
+        # We need to iterate over the class's inherited classes in reverse order
```

Comment:
That's pretty similar to what we've done. We're iterating over `c.__dict__.keys()` rather than `dir(c)` but the rest of your code looks the same. As above, I think we want to keep the reverse MRO loop.